module/VuFindSearch/src/VuFindSearch/Backend/EDS/SearchRequestModel.php

```diff
@@ -517,4 +549,43 @@ public function __set($property, $value)
 
         return $this;
     }
+
+    // all copied for draft purposes from Base/Params
+    protected $orFacets = [];
+    // protected function initFacetList($facetList, $facetSettings, $cfgFile = null)
+    public function initFacetList($facetList, $facetSettings, $config)
+    {
+        // $config = $this->configLoader
+        //     ->get($cfgFile ?? $this->getOptions()->getFacetsIni());
+        if (!isset($config->$facetList)) {
+            return false;
+        }
+        if (isset($config->$facetSettings->orFacets)) {
+            $orFields
+                = array_map('trim', explode(',', $config->$facetSettings->orFacets));
+        } else {
+            $orFields = [];
+        }
+        foreach ($config->$facetList as $key => $value) {
+            $useOr = (isset($orFields[0]) && $orFields[0] == '*')
+                || in_array($key, $orFields);
+            $this->addFacet($key, $value, $useOr);
+        }
+
+        return true;
+    }
+    public function addFacet($newField, $newAlias = null, $ored = false)
+    {
+        // if ($newAlias == null) {
+        //     $newAlias = $newField;
+        // }
+        // $this->facetConfig[$newField] = $newAlias;
+        if ($ored) {
+            $this->orFacets[] = $newField;
+        }
+    }
+    public function getFacetOperator($field)
+    {
+        return in_array($field, $this->orFacets) ? 'OR' : 'AND';
+    }
```

Comment:
So obviously I don't really want to duplicate all of this logic from Base/Params.  But I could not find a clean way to make the Params object available to SearchRequestModel.  Hoping for some advice here.